pkgs/development/python-modules/fastdtw/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cython
+, numpy
+}:
+
+buildPythonPackage rec {
+  pname = "fastdtw";
+  version = "0.3.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "2350fa6ec36bcad186eaf81f46eff35181baf04e324f522de8aeb43d0243f64f";
+  };
+
+  nativeBuildInputs = [
+    cython
+    numpy
+  ];
+
+  # Pypi's tarball doesn't contain tests
+  doCheck = false;
+  pythonImportsCheck = [ "fastdtw.fastdtw" ];
```

Comment:
please checkout from github, it contains tests

Reply:
I tried that earlier today, but had trouble with pytest finding the ``fastdtw`` modules. ``fastdtw`` cython-izes several modules, and pytest tries to find them in /build/source/fastdtw/..., but the cython-ized code is elsewhere. Would appreciate recommendations.

Reply:
https://github.com/NixOS/nixpkgs/pull/78868#issuecomment-580320472

Reply:
With github & test commit below (https://github.com/NixOS/nixpkgs/pull/78868/commits/1572d9699dd71b6aa1bbe2cddcfe89369ac4deb2)

```
============================= test session starts ==============================
platform linux -- Python 3.7.6, pytest-5.3.2, py-1.8.1, pluggy-0.13.1
rootdir: /build/source
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
____________________ ERROR collecting tests/test_fastdtw.py ____________________
ImportError while importing test module '/build/source/tests/test_fastdtw.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/unixccompiler.py:118: in _compile
    extra_postargs)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/ccompiler.py:910: in spawn
    spawn(cmd, dry_run=self.dry_run)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/spawn.py:36: in spawn
    _spawn_posix(cmd, search_path, dry_run=dry_run)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/spawn.py:159: in _spawn_posix
    % (cmd, exit_status))
E   distutils.errors.DistutilsExecError: command 'gcc' failed with exit status 1

During handling of the above exception, another exception occurred:
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/core.py:148: in setup
    dist.run_commands()
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/dist.py:966: in run_commands
    self.run_command(cmd)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/dist.py:985: in run_command
    cmd_obj.run()
/nix/store/amyz2nhkmqni69h8v923k7788kqjz2ci-python3.7-pytest-runner-5.2/lib/python3.7/site-packages/ptr.py:208: in run
    with self.project_on_sys_path():
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/contextlib.py:112: in __enter__
    return next(self.gen)
/nix/store/35q07v2ici22k9sgrzbq3js64zq4zal1-python3.7-setuptools-44.0.0/lib/python3.7/site-packages/setuptools/command/test.py:155: in project_on_sys_path
    self.run_command('build_ext')
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/cmd.py:313: in run_command
    self.distribution.run_command(command)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/dist.py:985: in run_command
    cmd_obj.run()
/nix/store/35q07v2ici22k9sgrzbq3js64zq4zal1-python3.7-setuptools-44.0.0/lib/python3.7/site-packages/setuptools/command/build_ext.py:84: in run
    _build_ext.run(self)
/nix/store/gqlb06xpa39i2l8z290ks6v1bm3an73c-python3.7-Cython-0.29.14/lib/python3.7/site-packages/Cython/Distutils/old_build_ext.py:186: in run
    _build_ext.build_ext.run(self)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/command/build_ext.py:340: in run
    self.build_extensions()
/nix/store/gqlb06xpa39i2l8z290ks6v1bm3an73c-python3.7-Cython-0.29.14/lib/python3.7/site-packages/Cython/Distutils/old_build_ext.py:195: in build_extensions
    _build_ext.build_ext.build_extensions(self)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/command/build_ext.py:449: in build_extensions
    self._build_extensions_serial()
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/command/build_ext.py:474: in _build_extensions_serial
    self.build_extension(ext)
/nix/store/35q07v2ici22k9sgrzbq3js64zq4zal1-python3.7-setuptools-44.0.0/lib/python3.7/site-packages/setuptools/command/build_ext.py:205: in build_extension
    _build_ext.build_extension(self, ext)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/command/build_ext.py:534: in build_extension
    depends=ext.depends)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/ccompiler.py:574: in compile
    self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/unixccompiler.py:120: in _compile
    raise CompileError(msg)
E   distutils.errors.CompileError: command 'gcc' failed with exit status 1

During handling of the above exception, another exception occurred:
setup.py:81: in <module>
    setup(**kwargs)
/nix/store/35q07v2ici22k9sgrzbq3js64zq4zal1-python3.7-setuptools-44.0.0/lib/python3.7/site-packages/setuptools/__init__.py:145: in setup
    return distutils.core.setup(**attrs)
/nix/store/aw85aijllnc5spk2fqbivygrhx6x7411-python3-3.7.6/lib/python3.7/distutils/core.py:163: in setup
    raise SystemExit("error: " + str(msg))
E   SystemExit: error: command 'gcc' failed with exit status 1

During handling of the above exception, another exception occurred:
tests/test_fastdtw.py:10: in <module>
    from fastdtw._fastdtw import fastdtw as fastdtw_c
E   ModuleNotFoundError: No module named 'fastdtw._fastdtw'
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.41s ===============================
```

Reply:
@jonringer tips please?

Reply:
python is trying to import the packages from the local directory. just move into the tests directory before running the tests